### PR TITLE
Emit click event for ui-shell nav items

### LINKED ISSUE #141: Implement Segment tracking events in Navbar links
### Feature description

After implementing the **[Documentation Dropdown](https://github.com/Qiskit/web-components/issues/100)** in the Navbar web component, we need to also ensure that all links send the appropriate Segment tracking events when clicked.

For this, we need to consider that now we'll be able to have one unified navbar / header for all Qiskit websites.

## Dependencies

This task depends on:

- https://github.com/Qiskit/web-components/issues/100

### Additional context

- We need to work closely with @agebbie to provide the data in the most useful format possible.
- We need to coordinate with @korgan00 to have a consistent approach to triggering tracking events here and in the widgets.

### PR DESCRIPTION
**Pull request by techtolentino**
## Changes

Fixes #141 

## Implementation details
- added a customEvent "onClick", so projects can attach their own click event (`@on-click="myFunction"`)
- the intention is to give consumers of the web-component, a way to do something based off of nav item clicks

For example:
```vue
<qiskit-ui-shell @on-click="myClickFunction"></qiskit-ui-shell>
```

## Screenshots

https://user-images.githubusercontent.com/6276074/190003310-772698bf-8f4d-44c0-b89d-126528d91e4e.mov

### CODE REVIEW

**Review by yoshiemuranaka — COMMENTED** [review]
really nice @techtolentino ✨ I like this overall approach. I think avoiding segment specific naming/properties makes a lot of sense. 
I left a few comments and a few additional thoughts... 
- Noticed that the Qiskit logo doesn't have the event attached, should it be included?
- Is there any way to demonstrate the on-click event in the storybook?
- I created a Segment interface in the settings file that can probably be deleted (pulled from previous navbar implementation)

File: components/ui-shell/index.ts [hunk 1]
```diff
@@ -190,11 +216,24 @@ export class QiskitUIShell extends LitElement {
...
       >
         ${userIcon} <span>Profile</span>
       </bx-side-nav-link>
     `;
   }
+
+  _handleClick = (label: string) => {
+    this.dispatchEvent(
+      new CustomEvent('onClick', {
+        detail: { label },
```
[the remark this answers is not in the document]
Comment (techtolentino) [reply components/ui-shell/index.ts:231]: I'm fine with this. @korgan00, what do you think?

[the remark this answers is not in the document]
Comment (korgan00) [reply components/ui-shell/index.ts:231]: Yeah, make sense to me.

File: components/ui-shell/index.stories.ts [hunk 2]
```diff
@@ -13,7 +13,12 @@ export default {
   title: 'UI Shell',
 };
 
-export const Default = () => html`<qiskit-ui-shell></qiskit-ui-shell>`;
+export const Default = () =>
+  html`<qiskit-ui-shell
+    @on-click="${(e: { detail: { label: string; url: string } }) => {
```
Comment (yoshiemuranaka) [inline components/ui-shell/index.stories.ts:18]: Could also use CustomEvent type to satisfy typescript

```suggestion
    @on-click="${(e: CustomEvent) => {
```

**Review by yoshiemuranaka — APPROVED** [review]
lgtm!

**Review by yoshiemuranaka — APPROVED** [review]
thanks @techtolentino !

**Pull request merged**